src/Aspire.Hosting/Dcp/Model/Container.cs

```diff
@@ -68,9 +68,26 @@ internal sealed class BuildContext
     [JsonPropertyName("args")]
     public List<EnvVar>? Args { get; set; }
 
+    // Optional build secret mounts to pass to the build command
+    [JsonPropertyName("secrets")]
+    public List<BuildContextSecret>? Secrets { get; set; }
+
     // Optional specific stage to use when building a multiple stage Dockerfile
     [JsonPropertyName("stage")]
     public string? Stage { get; set; }
+
+    // Optional additional tags to apply to the built image
+    [JsonPropertyName("tags")]
+    public List<string>? Tags { get; set; }
+}
+
+internal sealed class BuildContextSecret
+{
+    [JsonPropertyName("id")]
+    public string? Id { get; set; }
+
+    [JsonPropertyName("source")]
+    public string? Source { get; set; }
```

Comment:
What do we put in the `Source` field? Do we include the type?

Reply:
Added a comment to explain that this is a source filepath (same as source in a bind volume mount).

Reply:
@danegsta I think we need to talk through this bit a little bit more because I'm not quite sure how it is going to practically work. Consider the following code:

```csharp
var builder = DistributedApplication.CreateBuilder(args);

var nugettoken = builder.AddParameter("nugettoken", secret: true);

var container = builder.AddContainer("mycontainer", "myimage")
                       .WithDockerfile(contextPath)
                       .WithSecretBuildArg("nameOfSecret", nugettoken);
```

It seems to me that the source in the DCP Container model above is expecting a path to a file.  I think that we'll probably just want to give DCP the literal value and have it manage whether it either sets an environment variable to use with `--secret` or temporarily writes the secret value to the file.

Reply:
I think setting an ENV when you launch docker build would be preferrable to avoid any potential leakage to the filesystem.